### DISCUSSION TITLE
Multi-room audio select zone

### DISCUSSION THREAD

**Discussion by OliOp** [music-assistant/support#1057] [Feature requests and ideas]
Hello,

I have: 
- HA running on an RPI 4 (HA OS install)
- Logitech Media Server (squeezebox) running on an RPI 3 with a USB drive mounting at startup hosting my FLAC library and my playlists (LMS)
- a Soundavo WS66i (6 zone audio amp - hard wired speakers)
     - I've installed the HA/Soundavo integration. Works well for turning on/off zones, adjusting volume and selecting sources. 
     - Also comes with 2 integrated streamers (which can be connected to Spotify and the likes)
          - HA and streamers do not integrate well (or at all in terms of controls). I'm not using them for now.
- an iOS app (iPeng) with full control of the LMS server (playlists, favorites, etc.)

I'm trying to setup an e2e experience on HA to consume my FLAC library as well as a few radio stations and from time to time, a playlist from a search in a music provider db (spotify, youtube, iHeartRadio, etc) across my 6 audio zones.

I can mostly achieve the above with music assistant, mini media player, shell script to mount the USB drive to HA/media with the FLAC files.

Before HA... 
I would use iPeng (mobile app) to play audio sources to different RPIs (running PiCorePlayers with HiFiBerry DAC HATs).
Each PiCorePlayer was connected to an audio input of my Soundavo 6 zone amp.
I would control which source (which PiCorePlayer) would play in each zone using the Soundavo iOS app.
In short, iPeng to choose the songs, playlist, radio channels and the Soundavo app to assign sources to zones.

This works fine but I'm introducing HA in "my life"... another app with capabilities.

What I'd like... 
Be able to smoothly navigate and search my audio sources (local FLAC files, local playlists, music providers) and assign sources to zones.

I feel Music Assistant is checking the key boxes minus the **audio zone selection** (assigning sources (named "players" in Music Assistant) to audio zones (Soundavo amp)). In order to do this part in HA, I need to go to a dashboard and select the source via a card. Part of the out of the box Soundavo / HA integration.

It'd be great if I could select the audio zone within Music Assistant or somehow access the Music Assistant capabilities from a regular Dashboard/card.

Thanks for pointing me in the right direction if solutions already exists! 

Soundavo integration card
![image](https://user-images.githubusercontent.com/59370239/211200840-19c78bb6-ea2d-4189-b8ba-ca708ae3c63d.png)

**Comment by ryanwwest** [comment]
I'm curious about pretty much the same thing, though I don't have the Soundavo WS66i yet and am debating if it would be easier to replace it with 6 (or 12) Arylic Up2Stream DIY Amplifiers instead (but still control them with HA). Did you find a solution to easily assigning input sources to zones (possibly multiple zones for one source simultanously), from Music Assistant or another route? Since you mentioned Soundavo, I know they have an app to do this as well—did you or do you still rely on that and is it not sufficient if HA didn't work out?

**Reply by OzGav** [reply]
I’m not sure I understand the issue. Each “player” in MA has its own queue. So by its nature you are assigning input sources to zones. For example, you can start a Spotify playlist on player 1, another Spotify playlist on player 2 and an album from your local media on player 3? You can also group players and have one source sent to that,..

**Reply by ryanwwest** [reply]
Thanks for your quick reply. After reading other discussions on here, maybe what OP is asking is already available as it seems the 'zones' or 'players' aka various stereo speaker sets can already be grouped together (albeit maybe not very easy to edit groups on the fly, but it is possible). The only thing I'm not sure about is if you can have multiple sources being sent to multiple players at the same time, e.g. source 1 -> group 1 consisting of players 1 and 2, and source 2 -> group 2 consisting of players 3 and 4.

I'm also curious about the WS66i-specific questions from OP since I'm not sure if that's the best solution vs. cheaper DIY amps with HA support, but that's separate.

**Reply by OzGav** [reply]
Yes you can. Every “player” has its own queue (people have asked can you transfer queues between players which is a good idea but not possible yet)

The OP mentioned picoreplayers which work well with MA and might be a good solution for you depending on your use case

**Comment by OzGav** [comment]
Relates to v1

**Comment by lukaszzyla** [comment]
Hi,
I am in the same boat so I will try to explain further.
I have Dayton Audio 8 zone whole house audio system - DAX88.
It integrates to HA through rs232 and I can control zones through cards.
The players that are created in HA do not have the PLAY MEDIA feature when created - I do not know why.
Here you can see the players created with DAX/XANTECH integration:
<img width="733" alt="image" src="https://github.com/music-assistant/hass-music-assistant/assets/36972413/f9610417-34f9-4a78-b7b3-29995ba69904">

As you can see - one is different - have different features code.
When you decode them - they miss PLAY MEDIA function and thus are unavailable in MA and hard to use.
One player with lower features number is the only one you can see in spotify or use for direct media playing in HA, but MA does not pick it up as a possible player!
The others we control through cards - a card allows you to choose one of the configured sound inputs:
it looks like this:
![image](https://github.com/music-assistant/hass-music-assistant/assets/36972413/e582047b-3092-4e37-9264-10b6d8bc8d77)

I have LMS running on a VM and thes inputs you see are from LMS and USB-DAC.
I would also be very happy to have funcion enabled to control this kind of whole house audio systems.

What I am looking to accomplish is building an easy to use music player that is able to:
1. lock my inputs to certain media type (input RADIO only gives you access to few favourite stations)
2. Be able to easily control availability of source and create an easy to use interface for table to browse and play music (spotify/lms/etc). I have few of my sources running spotify connect so it is easy to choose a zone in spotify on your mobile, but then you have to go back to ha to your whole house audio card and choose where to play it.
3. So an interface combinign both (eg for each input independently) with a list of zones to play to would be of great use.
I have just came back to checking on MA as I was struggling to creat ANY player there after I installed my DAX88 integration.
So integration works in HA but doesnt really work in MA.

**Reply by lukaszzyla** [reply]
@OzGav

**Reply by OzGav** [reply]
I think you are pioneering this. I saw that you have found the previous disucssion about this device so that is all I know about it. Looking at the website it says this should have integrated Airplay. Does this show up for MA? I see it also has some digital connections so you could perhaps connect a Wiim Pro or similar to it?

**Comment by OliOp** [comment]
Hi folks,
Looks like Music Assistant V2 has OOTB support for what I was looking for back in 2023... Started playing with it again this weekend.

@ryanwwest 
Here's where I'm at... 
-Same amplifier (Soundavo), same sources as my initial post.
-I've reinstalled MASS

**Issue**: None of my Soundavo zones are available in the players' settings, still. I might be doing something wrong. 

Essentially, I'm back to where I was before for that issue.

**What changed?** 
I started using Lutron smart light switches and pico remotes. These remotes, to a certain extent, give me a better user experience. I have 1 pico remote per zone (these are physical devices working on an RF frequency/network).

Pico remotes allow me to consolidate the below functions:
-Turn on/off a Soundavo zone (toggle)
-Turn up/down the volume of a Soundavo zone
-Play/pause the LMS player (toggle)
-Skip to next track (LMS player)

**Controlling sources for zones**:
Although I could use HA Soundavo cards to control sources per zones, I prefer to use the Soundavo app. 

**Selecting local music (Playlists, albums, favorites):**
I could do this in HA (LMS integration/cards), I prefer to use iPeng (iOS app to control LMS)

**Muting a zone:**
The pico remotes only have 5 buttons. I did not want to use one for the mute. Instead, I use the "Play/Pause" (LMS player) which effictively pauses the stream for all zones using it. Not just the zone I want to mute.

At the end of the day, the experience I have in place now is somewhat better than before, but that is only thanks to the physical Pico remotes from Lutron. That said, in order for my Pico remotes to work as I desire, I need HA to assign actions to triggers and that works very well.  

We all have our preferences on how we control our media. In my case, I realized that I prefer to have a physical controller to perform the most basic tasks when listening to music (on/off, volume and play/pause (mute), skip) as opposed to have to open different apps on my phone.

A typical experience...
-Get up in the morning
-Turn on the zone I want (it has an automation that also sends the play command to LMS) pressing the button on the Pico remote
-I open my iPeng app on my iPhone and chose the music / stream I want to listen to 
If I get a phone call, I can promptly hit the Play/pause button on the Pico remote to immediatly mute the zone I'm in.

**Comment by kris10an** [comment]
Hi,
I have a similar situation with Google Chromecast audios and a +20 year old Russound 6-zone amplifier integrated with HA via RS232. Just got started with Music Assistant and was searching for solutions so came across this. 
I started thinking it could be possible to use the HA Universal Media Player integration to combine both the source streaming device (chromecast) and the amp zone (russound) into one player in HA and then expose that to MA. Haven't started testing yet but seems it could be a possibility as there is a lot of options with the Universal Media Player.

**Comment by lukaszzyla** [comment]
This is the solution I am currently using.
Created new universal media players with one source of audio and room speakers.
It is patch more than an elegan solution as devices create strange groups and there are some issues, but at least I have all my zones in MA now.
One of my zones here. Would be happy to see if you can think of more elegant and better working solution:

### ### ### MA Players ### ###
### ZONE: Master Bedroom ###
  - platform: universal
    name: MA_Master
    unique_id: MA_master
    media_content_type: music
    device_class: speaker
    announce: true
    children:
      - media_player.ma
    attributes:
      state: media_player.xantech8_master_bedroom
      volume_level: media_player.xantech8_master_bedroom|volume_level
    browse_media_entity: media_player.ma
    active_child_template: media_player.ma
    commands:
      toggle:
        service: media_player.toggle
        target:
          entity_id:
          - media_player.xantech8_master_bedroom
      turn_on:
        service: script.MA_Master_3
      turn_off:
        service: media_player.turn_off
        target:
          entity_id:
          - media_player.xantech8_master_bedroom
      volume_up:
        service: media_player.volume_up
        target:
          entity_id:
          - media_player.xantech8_master_bedroom
      volume_down:
        service: media_player.volume_down
        target:
          entity_id:
          - media_player.xantech8_master_bedroom
      volume_mute:
        service: media_player.volume_mute
        target:
          entity_id:
          - media_player.xantech8_master_bedroom
        data_template:
          entity_id: media_player.xantech8_master_bedroom
          is_volume_muted: "{{ not state_attr('media_player.xantech8_master_bedroom', 'is_volume_muted') }}"
      select_source:
        service: media_player.select_source
        target:
          entity_id:
          - media_player.xantech8_master_bedroom
        data:
          source: "MA"
      volume_set:
        service: media_player.volume_set
        target:
          entity_id:
          - media_player.xantech8_master_bedroom
        data:
          volume_level: "{{ volume_level }}"

________________________________
